eth2deposit/utils/intl.py

```diff
@@ -0,0 +1,95 @@
+import inspect
+import difflib
+from functools import reduce
+import json
+from typing import (
+    Any,
+    Dict,
+    Iterable,
+    List,
+    Mapping,
+    Sequence,
+)
+import os
+
+from eth2deposit.utils import config
+from eth2deposit.utils.constants import (
+    INTL_CONTENT_PATH,
+)
+from eth2deposit.exceptions import ValidationError
+
+
+def _get_from_dict(dataDict: Dict[str, Any], mapList: Iterable[str]) -> str:
+    '''
+    Iterate nested dictionaries
+    '''
+    try:
+        ans = reduce(dict.get, mapList, dataDict)
+        assert isinstance(ans, str)
+        return ans
+    except TypeError:
+        raise KeyError('%s not in internationalisation json file.' % mapList)
+    except AssertionError:
+        raise KeyError('The provided params (%s) were incomplete.' % mapList)
+
+
+def load_text(params: List[str], file_path: str='', func: str='', lang: str='') -> str:
+    '''
+    Determine and return the appropriate internationalisation text for a given set of `params`.
+    '''
+    if file_path == '':
+        # Auto-detect file-path based on call stack
+        file_path = inspect.stack()[1].filename
+        file_path = file_path[:-3] + '.json'     # replace .py with .json
+
+    if func == '':
+        # Auto-detect function based on call stack
+        func = inspect.stack()[1].function
+
+    if lang == '':
+        lang = config.language
+
+    # Determine path to json text
+    file_path_list = os.path.normpath(file_path).split(os.path.sep)
+    rel_path_list = file_path_list[file_path_list.index('eth2deposit') + 1:]
+    json_path = os.path.join(INTL_CONTENT_PATH, lang, *rel_path_list)
+
+    try:
+        # browse json until text is found
+        with open(json_path) as f:
+            text_dict = json.load(f)
+            return _get_from_dict(text_dict, [func] + params)
+    except (KeyError, FileNotFoundError):
+        # If text not found in lang, try return English version
+        if lang == 'en':
+            raise KeyError('%s not in %s file' % ([func] + params, json_path))
+        return load_text(params, file_path, func, 'en')
+
+
+def get_first_options(options: Mapping[str, Sequence[str]]) -> List[str]:
+    '''
+    Returns the first `option` in the values of the `options` dict.
+    '''
+    return list(map(lambda x: x[0], options.values()))
+
+
+def _closest_match(text: str, options: Iterable[str]) -> str:
+    '''
+    Finds the closest match to `text` in the `options_list`
+    '''
+    match = difflib.get_close_matches(text, options, n=1, cutoff=0.6)
```

Comment:
I really like how you use this method here! 👍 

Just FYI that it returns an answer even with the wrong input. e.g., input `12` gets `2: ελληνικά`.
```
Please choose your language ['1. العربية', '2. ελληνικά', '3. English', '4. Français', '5. Bahasa melayu', '6. Italiano', '7. 日本語', '8. 한국어', '9. Português do Brasil', '10. român', '11. 简体中文']:  [English]: 12
Επιλέξτε τη γλώσσα του μνημονικού σας ['1. 简体中文', '2. 繁體中文', '3. čeština', '4. English', '5. Italiano', '6. 한국어', '7. Português', '8. Español']:  [english]:
```

I think it's okay for our use case, just add a note here.

Reply:
My primary motivation behind this fuzzy matcher is not forcing people to have to be too precise will spelling mistakes or accents it also helps with unicode character differences (multiple ways of entering a visually similar character).

I could tune the cut-off parameter to reject this case, but the false negative rate goes up quickly then.